pkg/chatbot/twitch_chatbot.go

```diff
@@ -0,0 +1,116 @@
+package chatbot
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gempir/go-twitch-irc/v2"
+	"github.com/nicklaw5/helix/v2"
+	"github.com/saxypandabear/twitchsongrequests/internal/constants"
+	"github.com/saxypandabear/twitchsongrequests/pkg/db"
+	"github.com/zmb3/spotify/v2"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+)
+
+func SendChatMessage(broadcaster string, userStore db.UserStore, event *helix.EventSubChannelPointsCustomRewardRedemptionEvent) error {
+	u, err := userStore.GetUser(event.BroadcasterUserID)
+	if err != nil {
+		return err
+	}
+
+	chatClient := twitch.NewClient(broadcaster, "oauth:"+u.TwitchAccessToken)
+	chatClient.Join(broadcaster)
+
+	chatClient.OnConnect(func() {
+		spotifyToken := &oauth2.Token{
+			AccessToken:  u.SpotifyAccessToken,
+			RefreshToken: u.SpotifyRefreshToken,
+			Expiry:       *u.SpotifyExpiry,
+		}
+
+		spotifyConfig := &oauth2.Config{
+			ClientID:     os.Getenv(constants.SpotifyClientIDKey),
+			ClientSecret: os.Getenv(constants.SpotifyClientSecretKey),
+			Endpoint: oauth2.Endpoint{
+				AuthURL:  "https://accounts.spotify.com/authorize",
+				TokenURL: "https://accounts.spotify.com/api/token",
+			},
+		}
+
+		httpClient := spotifyConfig.Client(context.Background(), spotifyToken)
+		spotifyClient := spotify.New(httpClient)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		track, trackErr := getSpotifyTrack(ctx, spotifyClient, event.UserInput)
+		if trackErr != nil {
+			zap.L().Error("Error in Spotify search or retrieving the track", zap.Error(trackErr))
+			chatClient.Say(broadcaster, "There was an error searching for the song.")
+			return
+		}
+
+		if track != nil {
+			sendTrackMessage(chatClient, broadcaster, track, event.UserName)
+		} else {
+			chatClient.Say(broadcaster, "No matching song was found.")
+		}
+	})
+
+	err = chatClient.Connect()
+	if err != nil {
+		zap.L().Error("Error connecting to Twitch chat",
+			zap.String("id", event.BroadcasterUserID),
+			zap.String("broadcaster", broadcaster),
+			zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+func getSpotifyTrack(ctx context.Context, client *spotify.Client, input string) (*spotify.FullTrack, error) {
```

Comment:
This effectively reinvents the wheel for the Spotify code that I already have in the project, but without the learnings I had running it for a while. It would be nice to consolidate that code - maybe could pull it out of the `playerqueue.go` and make a more standalone Spotify utility file for these operations.